asset-transfer-basic/rest-api-typescript/src/server.ts

```diff
@@ -0,0 +1,67 @@
+import helmet from 'helmet';
+import { StatusCodes, getReasonPhrase } from 'http-status-codes';
+import express, { NextFunction, Request, Response } from 'express';
+import pino from 'pino';
+import pinoMiddleware from 'pino-http';
+
+const logger = pino({
+  level: process.env.LOG_LEVEL || 'info',
+});
+
+const app = express();
+const { BAD_REQUEST, INTERNAL_SERVER_ERROR, OK } = StatusCodes;
+
+app.use(pinoMiddleware({
+  logger,
+  customLogLevel: function customLogLevel(res, err) {
+    if (res.statusCode >= BAD_REQUEST && res.statusCode < INTERNAL_SERVER_ERROR) {
+      return 'warn';
+    }
+
+    if (res.statusCode >= INTERNAL_SERVER_ERROR || err) {
+      return 'error';
+    }
+
+    return 'debug';
+  },
+}));
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+if (process.env.NODE_ENV === 'development') {
+  // TBC
+}
+
+if (process.env.NODE_ENV === 'production') {
+  app.use(helmet());
+}
+
+app.get('/ready', (_req, res) => res.status(OK).json({
+  status: getReasonPhrase(OK),
+  timestamp: new Date().toISOString(),
+}));
+app.get('/live', (_req, res) => res.status(OK).json({
+  status: getReasonPhrase(OK),
+  timestamp: new Date().toISOString(),
+}));
+
+// TODO delete me
+app.get('/error', (_req, _res) => { throw new Error('arg'); });
+
```

Comment:
Add an OpenAPI?

https://github.com/hyperledgendary/fabric-rest-server/blob/7f25307e839abfd09a6583b14ec71737edebf7dd/rest-server/src/server.ts#L79

Reply:
I'd like to keep the sample to a bare (reasonable) minimum so that the Fabric SDK aspects don’t get too swamped with the rest, so I'm going to leave out OpenAPI for now. Can always add it later if it's a popular request.